### DISCUSSION TITLE
Product search for Admin area

### DISCUSSION THREAD

**Discussion by korberg** [opencart/opencart#12246] [Ideas]
Is there any particular reason why the search for administrator is quite useless - unless you will know the name of the product by heart?
As an Opencart maintainer and user I see that the autocomplete is not as useful as would be a global search for products. Especially when you have some 900+ products in the store.
So currently I have been able to modify the storefront search to be more useful than it is meant to be (including descriptions etc) and having found a product ID there i just edit a random products edit page's address to load the needed product I want to modify.
Needless to say, this is "sh*t", but I havent been able to find a way to produce a better search for admin without breaking it altogether.
So... If some stuff is in dev for OC4, I would recommend to address this issue.

**Comment by stalker780** [comment]
Agree.

Admin product search MUST use search model similar to frontend.
Split search string by words and search by partial %strings%:
https://github.com/opencart/opencart/blob/f31535a13a1d973965c669ac2e797efb48f128b1/upload/catalog/model/catalog/product.php#L69

Also there MUST be a filter by category and manufacturer.
Also quantity and price filters MUST be From:To.

Otherwise every single OC store is unusable and requires custom modifications for product list form and product autocompletes.

**Reply by stalker780** [reply]
I can prepare a pull request with this functionality. It is quite simple.
But all previous attempts to add correct filter by name or category were closed.
Every developer simply has an extension which adds needed modifications to core.
